dev_environment/e2e_test/setup_linux_dev_env.sh

```diff
@@ -0,0 +1,236 @@
+  wget https://go.dev/dl/$go_filename
+  tar -C /usr/local -xzf $go_filename && rm -f $go_filename
+  export PATH=$PATH:/usr/local/go/bin
+  echo export PATH=$PATH:/usr/local/go/bin >> /root/.bashrc
+
+if [ ! -d "/root/l3af-arch" ];
+then
+  git clone https://github.com/l3af-project/l3af-arch.git
+else
+  echo "/root/l3af-arch directory already exists"
+fi
+
+# Copy grafana configs and dashboards into place
+if [ -d "/var/lib/grafana/dashboards" ];
+then
+  echo "grafana directory already existed"
+else
+  mkdir -p /var/lib/grafana/dashboards
+fi
+chown grafana:grafana /var/lib/grafana/dashboards
+cp /root/l3af-arch/dev_environment/cfg/grafana/dashboards/*.json /var/lib/grafana/dashboards
+chown grafana:grafana /var/lib/grafana/dashboards/*.json
+cp /root/l3af-arch/dev_environment/cfg/grafana/provisioning/dashboards/l3af.yaml /etc/grafana/provisioning/dashboards
+chown root:grafana /etc/grafana/provisioning/dashboards/*.yaml
+cp /root/l3af-arch/dev_environment/cfg/grafana/provisioning/datasources/l3af.yaml /etc/grafana/provisioning/datasources
+chown root:grafana /etc/grafana/provisioning/datasources/*.yaml
+
+# Copy prometheus config and restart prometheus
+cp /root/l3af-arch/dev_environment/cfg/prometheus.yml /etc/prometheus/prometheus.yml
+if uname -a | grep -q 'WSL';
+then
+  echo "WSL DETECTED"
+  apt-get install daemon
+  /etc/init.d/prometheus-node-exporter stop
+  /etc/init.d/prometheus-node-exporter start
+
+  # Start and enable Grafana
+  sleep 1
+  /etc/init.d/grafana-server stop || true
+  /etc/init.d/grafana-server start || true
+else
+  # the configuration got copied, restart the prometheus service
+  systemctl daemon-reload
+  systemctl restart prometheus prometheus-node-exporter
+  systemctl enable prometheus.service
+  systemctl enable prometheus-node-exporter.service
+
+  # Start and enable Grafana
+  systemctl restart grafana-server
+  systemctl enable grafana-server.service
+fi
+
+# Get Linux source code to build our eBPF programs against
+if [ -d "/usr/src/linux" ];
+then
+  echo "Linux source code already exists, skipping download"
+else
+  git clone --branch v5.15 --depth 1 https://github.com/torvalds/linux.git /usr/src/linux
+fi
+
+LINUX_SRC_DIR=/usr/src/linux
+cd $LINUX_SRC_DIR
+make defconfig
+make prepare
+make headers_install
+
+if [ ! -d "/var/log/l3af" ];
+then
+  mkdir -p /var/log/l3af
+fi
+if [ ! -d "/var/l3afd" ];
+then
+  mkdir -p /var/l3afd
+fi
+
+BUILD_DIR=$LINUX_SRC_DIR/samples/bpf/
+
+# Where to store the tar.gz build artifacts
+BUILD_ARTIFACT_DIR=/srv/l3afd
+if [ ! -d "$BUILD_ARTIFACT_DIR" ];
+then
+  mkdir -p $BUILD_ARTIFACT_DIR
+fi
+
+cd $BUILD_DIR
+
+# Get the eBPF-Package-Repository repo containing the eBPF programs
+if [ ! -d "$BUILD_DIR/eBPF-Package-Repository" ];
+then
+  git clone https://github.com/l3af-project/eBPF-Package-Repository.git
+fi
+cd eBPF-Package-Repository
+
+# declare an array variable
```

Comment:
Some comments start with a capital letter, others don't.  Would be good to be consistent throughout this file.

Reply:
done